pkg/api/validation/validation.go

```diff
@@ -1022,11 +1022,19 @@ func ValidatePersistentVolumeClaim(pvc *api.PersistentVolumeClaim) field.ErrorLi
 func ValidatePersistentVolumeClaimUpdate(newPvc, oldPvc *api.PersistentVolumeClaim) field.ErrorList {
 	allErrs := ValidateObjectMetaUpdate(&newPvc.ObjectMeta, &oldPvc.ObjectMeta, field.NewPath("metadata"))
 	allErrs = append(allErrs, ValidatePersistentVolumeClaim(newPvc)...)
-	// if a pvc had a bound volume, we should not allow updates to resources or access modes
-	if len(oldPvc.Spec.VolumeName) != 0 {
+	// PVController needs to update PVC.Spec w/ VolumeName. Any other change is not allowed.
+	// Claims are immutable in order to enforce quota, range limits, etc. without gaming the system.
+	if len(oldPvc.Spec.VolumeName) == 0 {
+		// Only volumeName changes are allowed. Reset back to empty string after equality check
+		oldPvc.Spec.VolumeName = newPvc.Spec.VolumeName
```

Comment:
We definitely check at the lower level (storage will no-op a CAS on etcd where source and target are the same).  I can buy that we should have better info at the above level.

I think all no-op changes (for now) should be allowed down through validation, so as Tim notes the PR should treat a no-op as success regardless of value (I think you were going to do that based on the comment below, but just double checking).